metricflow/dataflow/nodes/window_reaggregation_node.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from typing import Sequence, Set
+
+from metricflow_semantics.dag.id_prefix import IdPrefix, StaticIdPrefix
+from metricflow_semantics.dag.mf_dag import DisplayedProperty
+from metricflow_semantics.specs.spec_classes import LinkableInstanceSpec, MetricSpec, TimeDimensionSpec
+from metricflow_semantics.visitor import VisitorOutputT
+
+from metricflow.dataflow.dataflow_plan import (
+    DataflowPlanNode,
+    DataflowPlanNodeVisitor,
+)
+from metricflow.dataflow.nodes.compute_metrics import ComputeMetricsNode
+
+
+class WindowReaggregationNode(DataflowPlanNode):
+    """A node that re-aggregates metrics using window functions.
+
+    Currently used for calculating cumulative metrics at various granularities.
+    """
+
+    def __init__(  # noqa: D107
+        self,
+        parent_node: ComputeMetricsNode,
+        metric_spec: MetricSpec,
+        order_by_spec: TimeDimensionSpec,
+        partition_by_specs: Sequence[TimeDimensionSpec],
```

Comment:
Realized the type for this should be `Sequence[InstanceSpec]` instead of `Sequence[TimeDimensionSpec]`. Fixed that higher up in the stack to avoid the rebasing, but will fix it here if I need to address feedback in this PR anyway.